### PR TITLE
[Enhancement] Add tracing util for FE

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

BE already added tracing framework, this PR adds tracing util for FE, and add some tracing for txn commit/publish

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 38 / 49 (77.55%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/TraceManager.java|13|23|56.52%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|12|13|92.31%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|12|12|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 38 / 49 (77.55%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/TraceManager.java|13|23|56.52%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|12|13|92.31%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|12|12|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1508,9 +1508,15 @@ public class Config extends ConfigBase {
...
      * this switch will be deleted after release the fqdn func
      */
     @ConfField(mutable = true)
     public static boolean enable_fqdn_func = false;
+
+    /**
+     * jaeger tracing endpoint, empty thing disables tracing
+     */
+    @ConfField
+    public static String jaeger_grpc_endpoint = "";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1521]: Can the jaeger trace the memory consumption at the peak?
If we configure the endpoint, does it indicate we push the trace log to the jaeger?

Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1521]: > Can the jaeger trace the memory consumption at the peak?
No, but if you can calculate the memory consumption at peak, you can add it to the span properties

> If we configure the endpoint, does it indicate we push the trace log to the jaeger?
Yes, the span data is in thrfit/protobuf format, and will be push to jaeger agent/server through udp/http/http2

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**